XenAdmin/Diagnostics/Checks/AssertCanEvacuateCheck.cs

```diff
@@ -184,6 +184,22 @@ private Problem GetProblem(IXenConnection connection, XenRef<VM> vmRef, string[]
 
                         return new NoPVDrivers(this, vm);
 
+                    case Failure.VM_LACKS_FEATURE:
+                        vm = connection.Resolve<VM>(vmRef);

```

Comment:
I see your point, I agree that would be better, but I think that shouldn't be part of a bugfix as it wouldn't be the minimal code change anymore - I only followed the existing pattern to avoid refactoring as much as possible.  Also, please note that the VM is not being resolved in the default case whereas it would be otherwise. 

I'd suggest we leave this as is (as part of this bugfix).